receiver/googlecloudspannerreceiver/internal/statsreader/databasereader.go

```diff
@@ -0,0 +1,85 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsreader
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/datasource"
+)
+
+type DatabaseReader struct {
+	database *datasource.Database
+	logger   *zap.Logger
+	readers  []Reader
+}
+
+func NewDatabaseReader(ctx context.Context,
+	databaseId *datasource.DatabaseId,
```

Comment:
You may want to find a better name for databaseId struct. Having databaseId.Id() is a bit odd.

Reply:
Will think about it